src/router.zig

```diff
@@ -108,98 +66,176 @@ pub fn Router(comptime Context: type, comptime routes: []const Route(Context)) R
+
+    const capture_args_info = function_info.args[3..];
+    var capture_arg_types: [capture_args_info.len]type = undefined;
+    for (capture_args_info) |arg, i| {
+        capture_arg_types[i] = arg.arg_type.?;
+    }
+    const CaptureArgs = std.meta.Tuple(&capture_arg_types);
+
+    const X = struct {
+        fn wrapper(ctx: Context, res: *Response, req: Request, params: []const trie.Entry) anyerror!void {
+            var capture_args: CaptureArgs = undefined;
+
+            std.debug.assert(params.len == capture_args.len); // Number of captures must equal the number of extra parameters in the Handler function
+
+            if (capture_args.len == 0) return handler(ctx, res, req);
+
+            comptime var arg_index = 0;
+            inline while (arg_index < capture_args.len) : (arg_index += 1) {
+                capture_args[arg_index] = switch (@TypeOf(capture_args[arg_index])) {
+                    []const u8 => params[arg_index].value,
+                    //?[]const u8 => if (params.len > 0) params[0].value else null,
+                    else => |ArgType| switch (@typeInfo(ArgType)) {
+                        .Int => std.fmt.parseInt(ArgType, params[arg_index].value, 10) catch {
+                            return res.notFound();
+                        },
+                        .Optional => |child| if (@typeInfo(child) == .Int)
+                            std.fmt.parseInt(ArgType, params[arg_index].value, 10) catch null
+                        else
+                            @compileError("Unsupported optional type " ++ @typeName(child)),
+                        else => @compileError("Unsupported type " ++ @typeName(ArgType)),
+                    },
+                };
+            }
+
+            return @call(.{}, handler, .{ ctx, res, req } ++ capture_args);
+        }
+    };
+
+    return X.wrapper;
+}
+
 /// Creates a builder namespace, generic over the given `Context`
 /// This makes it easy to create the routers without having to passing
 /// a lot of the types.
 pub fn Builder(comptime Context: type) type {
     return struct {
-        /// The generic handler type that all routes must match
-        pub const HandlerType = fn (Context, *Response, Request, ?*const anyopaque) anyerror!void;
-
-        /// Creates a new `Route` for the given HTTP Method that will be
-        /// triggered based on its path conditions
-        ///
-        /// When the path contains parameters such as ':<name>' it will be captured
-        /// and parsed into the type given as `capture_type`.
-        pub fn basicRoute(
-            comptime method: Request.Method,
-            comptime path: []const u8,
-            comptime CaptureType: ?type,
-            comptime handler: HandlerType,
-        ) Route(Context) {
-            return .{
-                .method = method,
+        const Handler = Route(Context).Handler;
+
+        pub fn get(path: []const u8, comptime handlerFn: anytype) Route(Context) {
```

Comment:
I'd like to provide any public-facing API to contain a doc comment describing what it does, especially as the auto-doc generation is being improved it will come in handy.

Reply:
Agreed, I'll go ahead and add doc comments